### ISSUE TITLE
In changing Dir, path not found

### ISSUE THREAD

**Issue by mindfuucker** [proofrock/pupcloud#15]
Pupcloud 0.7.2 

Something goes wrong when using the forward/backward buttons to go back to a folder. 
The console shows the following (redacted) 404 error:

```
# On first press back:
GET https://files.domain.nl/ls?path=Backups/fotoarchief/My%20full%20Name%20and%20Girl%20Friend%20Name%202015%20-%202030/2022/2022-04

# On second subsequent press back:
GET https://files.domain.nl//ls?path=Backups/fotoarchief/My%20full%20Name%20and%20Girl%20Friend%20Name%202015%20-%202030/2022

# On third subsequent press back:
GET https://files.domain.nl/ls?path=Backups/fotoarchief/My%20full%20Name%20and%20Girl%20Friend%20Name%202015%20-%202030
```
The file structure should look like this:
- Backups
   - fotoarchief
      - My full Name and Girl Friend Name 2015 - 2030
         - 2022
            - 2022-04

I'm not sure what goes wrong here, maybe the URL encoding/decoding goes wrong with the %2020 ?

**Comment by proofrock** [comment]
Ouch! Sorry about this. Will investigate in the weekend. Thanks for now!

**Comment by proofrock** [comment]
@mindfuucker I think I found the problem. Would you mind to test it, to ensure I don't mess up twice? If so (and thanks), please tell me your OS/arch of preference and I'll provide the binary.

**Comment by mindfuucker** [comment]
I'm using Linux AMD 64bit on my server. Might take until tomorrow to test it.

**Comment by proofrock** [comment]
Take your time, it's absolutely no problem 😃 and thanks! Here is a link: [https://we.tl/t-LwYVXQA9uX](https://we.tl/t-LwYVXQA9uX) The version is still 0.7.2, and I know I should be using pupcloud to share, but I don't have time to setup the reverse proxy. Next time 😄

**Comment by mindfuucker** [comment]
There is still a bug somewhere. It seems to be stuck on a directory. The URL-bar quickly changes and then changes back. It looks like you are registering a state twice. If I manually go skip a state and go back two states, then I reach the correct directory. 

How to reproduce:

1. Go to 2022-04 folder
2. Click back, you are now in 2022
3. Click back again: you are still in 2022 folder.

**Comment by proofrock** [comment]
Found, thanks. A stupid misplaced bracket. Try [this](https://we.tl/t-pgM6i2bXcE).

**Comment by proofrock** [comment]
> Could you add the directory path in the state-title

I was thinking the same thing. Would you care to open a separate issue?

**Comment by proofrock** [comment]
Hi @mindfuucker ! Did you have a chance to test it? If not, I'll consider it done, I used it for a while and it seems ok. Thanks again!

**Comment by proofrock** [comment]
Released in v0.8.0
